### ISSUE TITLE
Question about multiple loss.backward()? Need we add retain_graph=True?

### ISSUE THREAD

**Issue by FairyFali** [ruizheng20/robust_ticket#2]
I found in the code of `search_ticket.py`, it uses freelb adversarial training. So, it need to loss.backward() more than once (`adv_steps `steps in total), but the code doesn't use the parameter retain_graph. So, I suspect whether it will influence the optimization, like ignoring the gradients in first few steps (`adv_steps-1` steps) and only using the last backward gradient. 

I wonder whether there is such a problem and we should add retain_graph=True to avoid this problem? Hope to receive your reply.

**Comment by ruizheng20** [comment]
Thank you for following our work.

You are right, we use only the last adversarial example (the strongest one) to search for robust tickets. This is better for adversarial robustness. If you add retain_graph=True, it will be more beneficial for the accuracy on the test set.

I hope my answer is helpful!

**Comment by ruizheng20** [comment]
Thanks for your interest in our work.

You are right, we use only the last adversarial example (the strongest one) to search for robust tickets. This is better for adversarial robustness. If you add retain_graph=True, it will be more beneficial for the accuracy on the test set.

I hope my answer is helpful!

**Comment by FairyFali** [comment]
Thanks for your reply. It helps.
